graspologic/embed/ase.py

```diff
@@ -115,13 +116,13 @@ class AdjacencySpectralEmbed(BaseSpectralEmbed):
 
     def __init__(
         self,
-        n_components=None,
-        n_elbows=2,
-        algorithm="randomized",
-        n_iter=5,
-        check_lcc=True,
-        diag_aug=True,
-        concat=False,
+        n_components Optional[int] = None,
+        n_elbows: Optional[int] = 2,
+        algorithm: Optional[str] = "randomized",
```

Comment:
I'm not sure these are right - Optional specifically means that `None` is a valid value.  In these cases, we're using it to mean that the parameter provided has a default value if one is not specified - which is not what Optional means in a typing context.

Reply:
In other words, if the value always has to be an int AND present, then it shouldn't be Optional regardless of whether we set a default value for it.

If the value can be an int OR None, and we behave differently on None than we would if it were an int, then that would be an `Optional[int]` (or str, in this case).

Reply:
correct, I beleve `n_components` is the only one that is optional in this sense

Reply:
I believe I have a bad understanding of "Option", that's why I did this, it's more clear with your explanations @dwaynepryce

Reply:
Does that mean ''' n_elbows ''' for example 
will be just ''' n_elbows: int = 2 '''  ?

Reply:
correct!

Reply:
I just pushed a corrected version of that !

Reply:
I have a question on this - say I have some value `alpha` which is None by default, but can be given a `float`.
If `alpha` is left as `None`, it gets assigned a default float value.

Should the type hint for this be
`alpha: float = None` or
`alpha: Optional[float] = None`?

It seems from this discussion like it should be the first one, but the `mypy` linter is throwing an error when I set it up like this.

Reply:
Apologies for the resurrection, but in your example Alex it would be `Optional[float]` as `None` is a valid value. In addition, `None` is also the default but in this case the default isn't relevant to decide the Optional attribute or not.